src/Database/Connection.php

```diff
@@ -278,44 +280,74 @@ public function run(Query $query): StatementInterface
         return $this->getDisconnectRetry()->run(fn () => $this->_driver->run($query));
     }
 
+    /**
+     * Get query factory instance.
+     *
+     * @return \Cake\Database\QueryFactory
+     */
+    public function queryFactory(): QueryFactory
+    {
+        return $this->queryFactory ??= new QueryFactory($this);
+    }
+
     /**
      * Create a new SelectQuery instance for this connection.
      *
+     * @param \Cake\Database\ExpressionInterface|\Closure|array|string|float|int $fields Fields/columns list for the query.
+     * @param array|string $tables List of tables to query.
+     * @param array<string, string> $types Associative array containing the types to be used for casting.
      * @return \Cake\Database\Query\SelectQuery
      */
-    public function newSelectQuery(): SelectQuery
-    {
-        return new SelectQuery($this);
+    public function newSelectQuery(
+        ExpressionInterface|Closure|array|string|float|int $fields = [],
+        array|string $tables = [],
```

Comment:
Should we call this `from` instead of `tables`? It's how you would read/write the SQL

Reply:
That's a great idea. It makes the named parameter interface more like the underlying methods which is a nice memory aid.

Reply:
> Hrm. After doing all these changes and updating the tests I came to realize that the consistency of always having `table` or `table` was easier to operate than having `into` and `from` and more. In fact, I think we could just normalize on table and be better off.

The one place that's not true is joining on select queries.

Reply:
I agree that `table` doesn't feel right for SELECT queries since the `FROM` expression doesn't necessarily have to be a table.

Reply:
A subquery or CTE expression result is effectively a 'table' :wink:

Reply:
I'm ok with using table in our nomenclature. It allows for some more flexibility in other areas.

Reply:
I too can live with table :)